### ISSUE TITLE
[SemanticVersion] comparison with [Version] constructed with just major+minor

### ISSUE THREAD

**Issue by nightroman** [PowerShell/PowerShell#3786]
Tried packages:

- PowerShell-6.0.0-beta.1-win10-win2016-x64.zip
- PowerShell-6.0.0-beta.1-win81-win2012r2-x64.zip

This issue might be related to https://github.com/PowerShell/PowerShell/pull/3696

As far as I understand, expressions `$psversiontable.psversion -gt "3.0"` were fixed.

But the similar expressions `$psversiontable.psversion -gt ([version]"3.0")` do not work:

    Could not compare "6.0.0-beta" to "3.0". Error: "Cannot convert value "3.0" to type
    "System.Management.Automation.SemanticVersion". Error: "Cannot process argument because the value of argument
    "version" is not valid. Change the value of the "version" argument and run the operation again.""
    At line:1 char:1
    + $psversiontable.psversion -gt ([version]"3.0")
    + ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
        + CategoryInfo          : InvalidOperation: (:) [], RuntimeException
        + FullyQualifiedErrorId : ComparisonFailure

In other words, `[SemanticVersion]` comparison with `[Version]` constructed with just major+minor does not work.

**Comment by mklement0** [comment]
The simplest way to reproduce this:

```powershell
[System.Management.Automation.SemanticVersion] [version] '3.0'
```

The problem is that even though `[System.Management.Automation.SemanticVersion]` does have a constructor with `[version]` argument, it currently requires that that instance have a `.Build` component `> 0`.

The following therefore succeeds:

```powershell
[System.Management.Automation.SemanticVersion] [version] '3.0.0'
```

**Comment by iSazonov** [comment]
@mirichmo Should we reopen the Issue?
